clash-lib/src/Clash/Normalize/Transformations/EtaExpand.hs

```diff
@@ -63,15 +63,29 @@ etaExpansionTL (TransformContext is0 ctx) (Lam bndr e) = do
   e' <- etaExpansionTL ctx' e
   return $ Lam bndr e'
 
-etaExpansionTL (TransformContext is0 ctx) (Letrec xes e) = do
+etaExpansionTL (TransformContext is0 ctx) (Let (NonRec i x) e) = do
+  let ctx' = TransformContext (extendInScopeSet is0 i) (LetBody [i] : ctx)
+  e' <- etaExpansionTL ctx' e
+  case stripLambda e' of
+    (bs@(_:_),e2) -> do
+      let e3 = Let (NonRec i x) e2
+      changed (mkLams e3 bs)
+    _ -> return (Let (NonRec i x) e')
+ where
+  stripLambda (Lam bndr e0) =
+    let (bndrs,e1) = stripLambda e0
+     in (bndr:bndrs,e1)
+  stripLambda e' = ([],e')
```

Comment:
Create a top-level version so it can be shared between the NonRec and Rec clause.